docs/design_proposals/user_survey_hooks.md

```diff
@@ -0,0 +1,135 @@
+	return len(cfgs) > 1
+ },
+```
+
+### How to prompt users to take user surveys
+
+#### Rules to show survey prompt
+When prompting users with surveys, we need to keep in mind 2 important rules
+1) Don't prompt users too often.
+   Currently, we only prompt users to fill in HaTs survey every two weeks until they fill.
+2) Don't prompt users if they have already taken the survey.
+   Currently, we only prompt users to fill in Hats Survey if they haven't taken it in last 3 months.
+
+For non HaTs surveys or user surveys, we will follow the same rules,
+1) Prompt users to fill in the survey once 2 weeks if its relevant to them until they fill it.
+2) Stop prompting once they have taken the survey.
+
+The user survey information will be tracked in the existing `Survey` config in the skaffold global config
+```
+// SurveyConfig is the survey config information
+type SurveyConfig struct {
+	DisablePrompt *bool           `yaml:"disable-prompt,omitempty"`
+	LastTaken     string          `yaml:"last-taken,omitempty"`
+	LastPrompted  string          `yaml:"last-prompted,omitempty"`
++	UserSurveys []*UseSurvey `yaml:"user-surveys,omitempty"`
+}
+
+type UserSurvey struct {
+	ID              string `yaml:"id"`
+	Taken        *bool  `yaml:"taken,omitempty"`
+}
+```
+
+
+### Implementation details.
+The current `ShouldDisplaySurveyPrompt` returns true only if
+1) If survey prompt is not disabled and
+2) If Hats survey was not taken in last 3 months (stored in  `SurveyConfig.LastTaken`) and
+3) If survey prompt was not shown `SurveyConfig.LastPrompted` in last 2 weeks.
+
+This behavior will be changed to `ShouldDisplaySurveyPrompt` returns true if
+1) If survey prompt is not disabled and
+2)  If survey prompt was not shown `SurveyConfig.LastPrompted` in last 2 weeks and
+3) If there is an active relevant user survey is available or Hats survey was not taken in last 3 months
+
+If both active relevant user survey is available or Hats survey was not taken, then skaffold will give preference to user survey over hats survey.
+Since, hats survey never expire,
+- if user takes the user survey this time, hats survey will be prompted the next time.
+-  if the user never takes the user survey, the hats survey will be prompted once the user survey expires.
+   **Note User surveys should not run longer than a quarter so that we don't reduce the volume of Hats Survey,
+```
+pkg survey/config
+
+def init() {
+  for _, s := range surveys {
+     if s.id != hats.id {
+       if s.expiresAt.IsZero() || s.expiresAt.Sub(time.Now) > 90days
+          panic(fmt.Errorf("survey %s is running longer than 90 days. This will starve users from being shown the hats survey."))
+    }
+  }
+}
+
+```
+
+Alternate options:
+1) Another option is to show the Hats survey right after user survey. 
+   Look into if we can keep the same Hats response Form.
+
+## Metrics
+1) Measure sample responses to user surveys
+2) Monitor the number of Hats responses per month is not reducing.
```

Comment:
```suggestion
### Implementation details.
The current `ShouldDisplaySurveyPrompt` returns true only if
1) If survey prompt is not disabled and
2) If HaTS survey was not taken in last 3 months (stored in  `SurveyConfig.LastTaken`) and
3) If survey prompt was not shown `SurveyConfig.LastPrompted` in last 2 weeks.

This behavior will be changed to `ShouldDisplaySurveyPrompt` returns true if
1) If survey prompt is not disabled and
2)  If survey prompt was not shown `SurveyConfig.LastPrompted` in last 2 weeks and
3) If there is an active relevant user survey is available or HaTS survey was not taken in last 3 months

If both active relevant user survey is available or HaTS survey was not taken, then Skaffold will give preference to user survey over hats survey.
Since, HaTS survey never expire,
- if user takes the user survey this time, HaTS survey will be prompted the next time.
-  if the user never takes the user survey, the HaTS survey will be prompted once the user survey expires.
   **Note User surveys should not run longer than a quarter so that we don't reduce the volume of HaTS Survey,
```
pkg survey/config

def init() {
  for _, s := range surveys {
     if s.id != hats.id {
       if s.expiresAt.IsZero() || s.expiresAt.Sub(time.Now) > 90days
          panic(fmt.Errorf("survey %s is running longer than 90 days. This will starve users from being shown the hats survey."))
    }
  }
}

```

Alternate options:
1) Another option is to show the HaTS survey right after user survey. 
   Look into if we can keep the same HaTS response Form.

## Metrics
1) Measure sample responses to user surveys
2) Monitor the number of HaTS responses per month is not reducing.
```